### ISSUE TITLE
Form audit for required fields and labels

### ISSUE THREAD

**Issue by manfromjupyter** [nbgallery/nbgallery#639]
A number of forms are missing required fields as well as labels. Need either an aria label, aria-labelledby or encapsulated by a <label></label> tag

**Comment by manfromjupyter** [comment]
Would like to hold on the required field front of this until we do #848, else cannot really confirm the `required="required"`s are working other than a visual component of physically clicking in it then out.... it doesn't automatically catch all requireds when trying to submit form
